src/base/bittorrent/customstorage.cpp

```diff
@@ -49,7 +49,33 @@ CustomStorage::CustomStorage(const lt::storage_params &params, lt::file_pool &fi
 
 bool CustomStorage::verify_resume_data(const lt::add_torrent_params &rd, const lt::aux::vector<std::string, lt::file_index_t> &links, lt::storage_error &ec)
 {
-    const QDir saveDir {m_savePath};
+    handleCompleteFiles(m_savePath);
+    return lt::default_storage::verify_resume_data(rd, links, ec);
+}
+
+void CustomStorage::set_file_priority(lt::aux::vector<lt::download_priority_t, lt::file_index_t> &priorities, lt::storage_error &ec)
+{
+    m_filePriorities = priorities;
+    lt::default_storage::set_file_priority(priorities, ec);
+}
+
+lt::status_t CustomStorage::move_storage(const std::string &savePath, lt::move_flags_t flags, lt::storage_error &ec)
+{
+    const QString newSavePath {Utils::Fs::expandPathAbs(QString::fromStdString(savePath))};
+
+    if (flags == lt::move_flags_t::always_replace_files)
+        handleCompleteFiles(newSavePath);
+
+    const lt::status_t ret = lt::default_storage::move_storage(savePath, flags, ec);
+    if (ret != lt::status_t::fatal_disk_error)
+        m_savePath = newSavePath;
+
+    return ret;
+}
+
+void CustomStorage::handleCompleteFiles(const QString &savePath)
+{
+    const QDir saveDir {savePath};
 
     const lt::file_storage &fileStorage = files();
     for (const lt::file_index_t fileIndex : fileStorage.file_range()) {
```

Comment:
Not related to the PR, just seeking for clarification. About this part:
```c++
if ((m_filePriorities.end_index() > fileIndex) && (m_filePriorities[fileIndex] == lt::dont_download))
    continue;
```
Is this check `(m_filePriorities.end_index() > fileIndex)` necessary? Also would it be better to have the following?
```c++
if ((fileIndex > m_filePriorities.end_index()) || (m_filePriorities[fileIndex] == lt::dont_download))
    continue;
```

Reply:
>Is this check `(m_filePriorities.end_index() > fileIndex)` necessary?

IIRC, libtorrent accepts priorities array with the size different than files count. In case of lesser count all missing elements are considered as default priority.